### ISSUE TITLE
Maximum dimensions allowed

### ISSUE THREAD

**Issue by sc28** [naught101/sobol_seq#15]
Hi, it seems the current implementation of sobol_seq allows generating sequences of coordinates for up to 40 dimensions.

I.e. executing the function for 100 dimensions returns the message:
```
vec, seed = sobol_seq.i4_sobol(100, 1)
I4_SOBOL - Fatal error!
  The spatial dimension DIM_NUM should satisfy:
    1 <= DIM_NUM <= 40
  But this input value is DIM_NUM = 100
```

On the [linked page](http://people.sc.fsu.edu/~jburkardt/py_src/sobol/sobol.html) towards the original Sobol library, it states that there are improved implementations allowing up to 1111 dimensions. **Are there any intentions of adopting those improvements here?**

Thanks in any case for this sobol_seq implementation, very convenient!

**Comment by naught101** [comment]
This library is basically in maintenance mode at the moment - I don't really need it for my work, and I don't have a lot of time for it. But I'm happy to take pull requests, if you want to have a go at re-implementing those improvements here.

**Comment by fazaghifari** [comment]
Hi @sc28 , I've created a function for higher dimension Sobol. I translated the function from the original c++ program https://web.maths.unsw.edu.au/~fkuo/sobol/. You can find the function here https://github.com/fazaghifari/SobolSampling .  However, I have no idea and I don't have a lot of time to combine my function with this repo.

**Comment by tupuinui** [comment]
FYI @fazaghifari and @naught101, I have a PR on SciPy to include Sobol'. It fixes the zero (issue present here but not in SobolSampling) and allow dimensions up to 21201. https://github.com/scipy/scipy/pull/10844

**Comment by naught101** [comment]
@tupuinui Cool, nice work. Will that PR include equivalent functionality to all of this module? 'cause if it does, I'll be happy to archive this and point people to scipy instead. If so, could you please keep us up to date on the progress?

**Comment by tupuinui** [comment]
@naught101 Thanks 😄. I checked and yes all the functionalities seemed to be there: random number generation using uniform or normal PDF, prime numbers, scaling, iterative design. I will let you know as soon as this gets merged into SciPy!

**Comment by naught101** [comment]
@tupuinui looks like this is merged now? Nice work - that looks like a huge PR.

I'm thinking I should deprecate this module and leave a link on the readme pointing to http://scipy.github.io/devdocs/generated/scipy.stats.qmc.Sobol.html, does that make sense to you?

**Comment by tupuinui** [comment]
@naught101 Thanks. Indeed it is merged and will be in scipy 1.7. So yes if you don't plan to add other functionalities to your package, it would make sense to redirect to scipy. Thanks for that.

**Comment by naught101** [comment]
OK, done. Thanks for your work @tupuinui :)
